src/AutoMapper/TypeMap.cs

```diff
@@ -263,7 +263,7 @@ public void Seal(TypeMapRegistry typeMapRegistry, IConfigurationProvider configu
                     .Union(_inheritedMaps)
                     .OrderBy(map => map.MappingOrder).ToArray();
 
-            MapExpression = TypeMapPlanBuilder.BuildMapperFunc(this, configurationProvider, typeMapRegistry);
+            MapExpression = new TypeMapPlanBuilder(configurationProvider, typeMapRegistry, this).CreateMapperLambda();

```

Comment:
I say put back TypeMapPlanBuilder.BuildMapperFunc and have it just implement this line.  Also make constructor and function call private, so that there's no way someone could accidentally use TypeMapPlanBuilder incorrectly.